archive-tar.c

```diff
@@ -463,18 +463,34 @@ static int write_tar_filter_archive(const struct archiver *ar,
 	filter.use_shell = 1;
 	filter.in = -1;
 
-	if (start_command(&filter) < 0)
-		die_errno(_("unable to start '%s' filter"), argv[0]);
-	close(1);
-	if (dup2(filter.in, 1) < 0)
-		die_errno(_("unable to redirect descriptor"));
-	close(filter.in);
+	if (!strcmp("gzip -cn", ar->data)) {
+		char outmode[4] = "wb6\0";
```

Comment:
Why not `"wb\0\0"`? The previous code did not default to compression level 6, after all, but let zlib decided...

Reply:
Better `"wb\0"`, actually, as that automatically NUL terminates, i.e. it is already 4 bytes long.